pkg/storegateway/bucket.go

```diff
@@ -94,9 +94,9 @@ type BucketStore struct {
 	snapshotter          *indexheader.Snapshotter
 	snapshotterStartOnce sync.Once
 
-	// Sets of blocks that have the same labels. They are indexed by a hash over their label set.
-	blocksMx sync.RWMutex
-	blocks   map[ulid.ULID]*bucketBlock
+	// Sets of blocks that have the same labels. The blocksMx mutex synchronizes the data in both sets.
+	blocksMx sync.Mutex
```

Comment:
If this `Mutex` is only used for `blockSet` now, can you rename it to `blockSetMx` and move it to be next to `blockSet`?